AdminCore.Common/Authorization/AdminCoreRolesPolicy.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AdminCore.Common.Interfaces;
+using AdminCore.Constants;
+using AdminCore.Constants.Enums;
+using AdminCore.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.Options;
+
+namespace AdminCore.Common.Authorization
+{
+    public class AdminCoreRolesPolicy : DefaultAuthorizationPolicyProvider
+    {
+        private readonly IConfiguration _configuration;
+        private readonly AuthorizationOptions _options;
+
+        public AdminCoreRolesPolicy(IOptions<AuthorizationOptions> options, IConfiguration configuration) : base(options)
+        {
+            _configuration = configuration;
+            _options = options.Value;
+        }
+
+        public override async Task<AuthorizationPolicy> GetPolicyAsync(string policyName)
```

Comment:
is this tested?

Reply:
I don't think this is really that testable given that base.GetPolicyAsync calls
```
Task.FromResult<AuthorizationPolicy>(this._options.GetPolicy(policyName));
```

Which relies on policies being registered with the Authorization middleware I think?

Reply:
When unit testing functionality of `AdminCoreRolesPolicy` class you shouldn't care what the internal behaviour of `base.GetPolicyAsync` is, just mock the return. Consider revising your code and make it testable, if that's not possible then fine, we will need to rely on manual tests, perhaps consider documenting manual test scenarios you've tried on this PR. There are some scenarios here I think we should test, like null handling, employee roles and policy building.